File: HNReaderObjC/HNReaderObjC/StoriesFetchRequest.m
//
//  StoriesFetchRequest.m
//  HNReaderObjC
//
//  Created by Gorceag Olga on 13/11/16.
//  Copyright © 2016 Gorceag Olga. All rights reserved.
//

#import "StoriesFetchRequest.h"

NSString *const ItemPath = @"v0/item/%@.json";
NSString *const TopStoriesPath = @"v0/topstories.json";

@interface StoriesFetchRequest ()

@property (nonatomic, strong) NSArray *itemsPathsToFetch;
@property (nonatomic, strong) NSArray *idsToFetchArary;

@end

@implementation StoriesFetchRequest

- (void)setIDsToFetch:(NSArray *)idsToFetch {
    NSMutableArray *pathsArray = [NSMutableArray array];
    for (NSNumber *number in idsToFetch) {
        [pathsArray addObject:[NSString stringWithFormat:ItemPath, number]];
    }
    self.itemsPathsToFetch = pathsArray;
    self.idsToFetchArary = idsToFetch;
}
@end

File: HNReaderSwift/HNReaderSwift/StoriesFetchRequest.swift
//
//  StoriesFetchRequest.swift
//  HNReaderSwift
//
//  Created by Gorceag Olga on 16/11/16.
//  Copyright © 2016 Gorceag Olga. All rights reserved.
//

import UIKit

let ItemPath = "v0/item/%@.json"
let TopStoriesPath = "v0/topstories.json";

class StoriesFetchRequest: NSObject {

    private(set) var itemsPathsToFetchArray: Array<String> = Array()
    private(set) var itemsIDsToFetchArray: Array<NSNumber>?

    public func setItemsIDsToFetch(itemsIDsArray: Array<NSNumber>) {
        itemsPathsToFetchArray.removeAll()
        for number in itemsIDsArray {
            itemsPathsToFetchArray.append(String(format: ItemPath, number))
        }
        itemsIDsToFetchArray = itemsIDsArray
    }

}
